File: drawee-backends/drawee-pipeline/src/main/java/com/facebook/drawee/backends/pipeline/info/ImageLoadStatus.java
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
package com.facebook.drawee.backends.pipeline.info;

import static com.facebook.drawee.backends.pipeline.info.ImageLoadStatus.CANCELED;
import static com.facebook.drawee.backends.pipeline.info.ImageLoadStatus.ERROR;
import static com.facebook.drawee.backends.pipeline.info.ImageLoadStatus.INTERMEDIATE_AVAILABLE;
import static com.facebook.drawee.backends.pipeline.info.ImageLoadStatus.ORIGIN_AVAILABLE;
import static com.facebook.drawee.backends.pipeline.info.ImageLoadStatus.REQUESTED;
import static com.facebook.drawee.backends.pipeline.info.ImageLoadStatus.SUCCESS;
import static com.facebook.drawee.backends.pipeline.info.ImageLoadStatus.UNKNOWN;
import static java.lang.annotation.RetentionPolicy.SOURCE;

import androidx.annotation.IntDef;
import java.lang.annotation.Retention;

@Retention(SOURCE)
@IntDef({
  UNKNOWN,
  REQUESTED,
  ORIGIN_AVAILABLE,
  INTERMEDIATE_AVAILABLE,
  SUCCESS,
  CANCELED,
  ERROR,
})
public @interface ImageLoadStatus {

  int UNKNOWN = -1;
  int REQUESTED = 0;
  int ORIGIN_AVAILABLE = 1;
  int INTERMEDIATE_AVAILABLE = 2;
  int SUCCESS = 3;
  int CANCELED = 4;
  int ERROR = 5;
}

File: ui-common/src/main/java/com/facebook/fresco/ui/common/ImageLoadStatus.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.fresco.ui.common

enum class ImageLoadStatus(val value: Int) {
  UNKNOWN(-1),
  REQUESTED(0),
  INTERMEDIATE_AVAILABLE(2),
  SUCCESS(3),
  ERROR(5),
  EMPTY_EVENT(7),
  RELEASED(8);

  /**
   * This was probably only used in open source version, so we might be able to remove this custom
   * string mapping.
   */
  override fun toString(): String {
    return when (this) {
      REQUESTED -> "requested"
      SUCCESS -> "success"
      INTERMEDIATE_AVAILABLE -> "intermediate_available"
      ERROR -> "error"
      RELEASED -> "released"
      else -> "unknown"
    }
  }

  companion object {
    private val VALUES = values()

    fun fromInt(value: Int): ImageLoadStatus? = VALUES.firstOrNull { it.value == value }
  }
}
